src/library/basesqltablemodel.cpp

```diff
@@ -842,7 +842,12 @@ void BaseSqlTableModel::setTrackValueForColumn(TrackPointer pTrack, int column,
     } else if (fieldIndex(ColumnCache::COLUMN_LIBRARYTABLE_TRACKNUMBER) == column) {
         pTrack->setTrackNumber(value.toString());
     } else if (fieldIndex(ColumnCache::COLUMN_LIBRARYTABLE_LOCATION) == column) {
-        pTrack->setLocation(value.toString());
+        DEBUG_ASSERT_AND_HANDLE(pTrack->getLocation() == value.toString()) {
+            qWarning() << "The location of track cannot be modified from"
+                    << pTrack->getLocation()
+                    << "to"
+                    << value.toString();
+        }
```

Comment:
I will add an else block with a DEBUG_ASSERT_AND_HANDLE macro that prints a 
warning when trying to edit an unsupported column:
- COLUMN_LIBRARYTABLE_LOCATION
- COLUMN_LIBRARYTABLE_FILETYPE
- COLUMN_LIBRARYTABLE_DURATION
- COLUMN_LIBRARYTABLE_BITRATE
- ...and all other columns that are not handled in the if/else before

On 09/20/2015 06:06 PM, Daniel Schürmann wrote:
>
> In src/library/basesqltablemodel.cpp 
> <https://github.com/mixxxdj/mixxx/pull/718#discussion_r39931353>:
>
> > @@ -842,7 +842,12 @@ void BaseSqlTableModel::setTrackValueForColumn(TrackPointer pTrack, int column,
> >      } else if (fieldIndex(ColumnCache::COLUMN_LIBRARYTABLE_TRACKNUMBER) == column) {
> >          pTrack->setTrackNumber(value.toString());
> >      } else if (fieldIndex(ColumnCache::COLUMN_LIBRARYTABLE_LOCATION) == column) {
> > -        pTrack->setLocation(value.toString());
> > +        DEBUG_ASSERT_AND_HANDLE(pTrack->getLocation() == value.toString()) {
> > +            qWarning() << "The location of track cannot be modified from"
> > +                    << pTrack->getLocation()
> > +                    << "to"
> > +                    << value.toString();
> > +        }
>
> IMHO we can be more strict here and assert if one tries to modify the 
> location.
> DEBUG_ASSERT_AND_HANDLE(false)
> This would be called if a user edits the location column.
> Why would a user do that?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/718/files#r39931353>.
>